src/hash_builder/mod.rs

```diff
@@ -227,33 +249,34 @@ impl HashBuilder {
 
             // Adjust the state masks for branch calculation
             let extra_digit = current[len];
-            if self.state_masks.len() <= len {
-                let new_len = len + 1;
-                trace!(target: "trie::hash_builder", new_len, old_len = self.state_masks.len(), "scaling state masks to fit");
-                self.state_masks.resize(new_len, TrieMask::default());
+
+            let new_len = len + 1;
+            if state_len < new_len {
```

Comment:
this adds confusion, and AFAIK modern assembly has `JLE` instruction that does this comparison in one go